src/Q1.c

```diff
@@ -61,20 +64,34 @@ void * thread_func(void *arg){
     SetBit(places, place);
     pthread_mutex_unlock(&mut); 
 
-    // always some place available
-    printRegister(elapsedTime(), threadi, getpid(), pthread_self(), dur, place, ENTER);
+    // sending message with place to private fifo
+    char sendMessage[BUFSIZE];
+
+    if (elapsedTime() + dur <= nsecs) {
+        // always some place available
+        sprintf(sendMessage,"[ %d, %d, %ld, %d, %d ]", threadi, getpid(), pthread_self(), dur, place);
+        printRegister(elapsedTime(), threadi, getpid(), pthread_self(), dur, place, ENTER);
+    }
+
+    else {
+        closed.x=1;
```

Comment:
este valor só pode ser alterado de 0 para 1, mas como é uma variável global que pode ser acedida por todas as threads devia ter um mutex
```suggestion
        pthread_mutex_lock(&mut2);   // necessário criar outro mutex
        closed.x=1;
        pthread_mutex_unlock(&mut2);  
```